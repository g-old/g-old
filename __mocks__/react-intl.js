// ./__mocks__/react-intl.js
import React from 'react';

const Intl = require.requireActual('react-intl');
const identityFunction = i => i;

// Here goes intl context injected into component, feel free to extend
const intl = {
  formatMessage: ({ defaultMessage }) => defaultMessage,
  formatDate: identityFunction,
  formatTime: identityFunction,
  formatRelative: identityFunction,
  formatNumber: identityFunction,
  formatPlural: identityFunction,
  formatHTMLMessage: identityFunction,
  now: identityFunction,
};

Intl.injectIntl = Node => {
  const renderWrapped = props => <Node {...props} intl={intl} />;
  renderWrapped.displayName = Node.displayName || Node.name || 'Component';
  return renderWrapped;
};
/* eslint-disable react/prop-types */
Intl.FormattedMessage = ({ defaultMessage }) => <span>{defaultMessage}</span>;
Intl.FormattedHTMLMessage = ({ defaultMessage }) => (
  <span>{defaultMessage}</span>
);
/* eslint-enable react/prop-types */

module.exports = Intl;
/*
import React, { Component } from 'react';

const Intl = require('react-intl');

const identityFunction = i => i;

// Here goes intl context injected into component, feel free to extend
const intl = {
  formatDate: identityFunction,
  formatTime: identityFunction,
  formatRelative: identityFunction,
  formatNumber: identityFunction,
  formatPlural: identityFunction,
  formatHTMLMessage: identityFunction,
  now: identityFunction,
  formatMessage: o => o.defaultMessage,
};

// Mock following implementations so that they do not break
Intl.injectIntl = Node => {
  // If we are injecting a Class, return a class, otherwise return the functional component
  // This is useful in cases where you are shallowing a component to test a single method.
  if (Node.prototype instanceof Component) {
    return class extends Node {
      static defaultProps = {
        ...(Node.defaultProps || {}),
        intl,
      };

      render() {
        return <Node {...this.props} />;
      }
    };
  }
  return props => <Node {...props} intl={intl} />;
};

Intl.FormattedMessage = ({ defaultMessage }) => <span>{defaultMessage}</span>;
Intl.FormattedHTMLMessage = ({ defaultMessage }) => (
  <span>{defaultMessage}</span>
);

module.exports = Intl;
*/
