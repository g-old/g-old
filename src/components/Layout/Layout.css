/**
 * React Starter Kit (https://www.reactstarterkit.com/)
 *
 * Copyright Â© 2014-present Kriasoft, LLC. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

@import '../variables.css';

/*
 * normalize.css is imported in JS file.
 * If you import external CSS file from your internal CSS
 * then it will be inlined and processed with CSS modules.
 */

/*
 * Base styles
 * ========================================================================== */

html {
  color: #222;
  font-weight: 100;
  font-size: 1em; /* ~16px; */
  font-family: var(--font-family-base);
  line-height: 1.375; /* ~22px */

  /* to disable scroll jumping
  https://aykevl.nl/2014/09/fix-jumping-scrollbar
  sideeffect: no horizontal scrolling possible
  */
  width: 100vw;
  overflow-x: hidden;
}

.layout {
  min-height: 100vh;
  position: relative;
}

.content {
  margin: 0 auto;
  padding: 0 0 60px;
  max-width: var(--max-content-width);
}

.footer {
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;
}

body {
  margin: 0;
}

a {
  color: #0074c2;
  text-decoration: none;
}

strong {
  font-weight: 600;
}

/*
 * Remove text-shadow in selection highlight:
 * https://twitter.com/miketaylr/status/12228805301
 *
 * These selection rule sets have to be separate.
 * Customize the background color to match your design.
 */

::-moz-selection {
  background: #b3d4fc;
  text-shadow: none;
}

::selection {
  background: #b3d4fc;
  text-shadow: none;
}

/*
 * A better looking default horizontal rule
 */

hr {
  display: block;
  height: 1px;
  border: 0;
  border-top: 1px solid #ccc;
  margin: 1em 0;
  padding: 0;
}

/*
 * Remove the gap between audio, canvas, iframes,
 * images, videos and the bottom of their containers:
 * https://github.com/h5bp/html5-boilerplate/issues/440
 */

audio,
canvas,
iframe,
img,
svg,
video {
  vertical-align: middle;
}

/*
 * Remove default fieldset styles.
 */

fieldset {
  border: 0;
  margin: 0;
  padding: 0;
}

/*
 * Allow only vertical resizing of textareas.
 */

textarea {
  resize: vertical;
}

/*
 * Browser upgrade prompt
 * ========================================================================== */

:global(.browserupgrade) {
  margin: 0.2em 0;
  background: #ccc;
  color: #000;
  padding: 0.2em 0;
}

/*
 * Print styles
 * Inlined to avoid the additional HTTP request:
 * http://www.phpied.com/delay-loading-your-print-css/
 * ========================================================================== */

@media print {
  *,
  *::before,
  *::after {
    background: transparent !important;
    color: #000 !important; /* Black prints faster: http://www.sanbeiji.com/archives/953 */
    box-shadow: none !important;
    text-shadow: none !important;
  }

  a,
  a:visited {
    text-decoration: underline;
  }

  a[href]::after {
    content: ' (' attr(href) ')';
  }

  abbr[title]::after {
    content: ' (' attr(title) ')';
  }

  /*
   * Don't show links that are fragment identifiers,
   * or use the `javascript:` pseudo protocol
   */

  a[href^='#']::after,
  a[href^='javascript:']::after {
    content: '';
  }

  pre,
  blockquote {
    border: 1px solid #999;
    page-break-inside: avoid;
  }

  /*
   * Printing Tables:
   * http://css-discuss.incutio.com/wiki/Printing_Tables
   */

  thead {
    display: table-header-group;
  }

  tr,
  img {
    page-break-inside: avoid;
  }

  img {
    max-width: 100% !important;
  }

  p,
  h2,
  h3 {
    orphans: 3;
    widows: 3;
  }

  h2,
  h3 {
    page-break-after: avoid;
  }
}

.loader {
  height: 4px;
  width: 100%;
  position: fixed;
  overflow: hidden;
}

.loader::before {
  display: block;
  position: absolute;
  content: '';
  left: -200px;
  width: 200px;
  height: 4px;
  background-color: #540b97;
  animation: loading 2s linear infinite;
}

@keyframes loading {
  from {
    left: -200px;
    width: 30%;
  }

  50% {
    width: 30%;
  }

  70% {
    width: 70%;
  }

  80% {
    left: 50%;
  }

  95% {
    left: 120%;
  }

  to {
    left: 100%;
  }
}

.banner {
  overflow: hidden;

  img {
    min-width: 32rem;
    max-width: 100%;
    display: block;
    margin: 0 auto;
  }
}
