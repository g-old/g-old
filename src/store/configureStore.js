import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from '../reducers';
import createHelpers from './createHelpers';
import createLogger from './logger';
// import { SESSION_LOGOUT_SUCCESS } from '../constants';

export default function configureStore(initialState, helpersConfig) {
  const helpers = createHelpers(helpersConfig);
  const middleware = [thunk.withExtraArgument(helpers)];

  let enhancer;

  if (__DEV__) {
    middleware.push(createLogger());

    // https://github.com/zalmoxisus/redux-devtools-extension#redux-devtools-extension
    let devToolsExtension = f => f;
    if (process.env.BROWSER && window.devToolsExtension) {
      devToolsExtension = window.devToolsExtension();
    }

    enhancer = compose(
      applyMiddleware(...middleware),
      devToolsExtension,
    );
  } else {
    enhancer = applyMiddleware(...middleware);
  }
  /*
  const rootReducer = (state, action) => {
    if (action.type === SESSION_LOGOUT_SUCCESS) {
      // eslint-disable-next-line no-param-reassign
      state = undefined;
    }
    return appReducer(state, action);
  };
*/
  // See https://github.com/rackt/redux/releases/tag/v3.1.0
  const store = createStore(rootReducer, initialState, enhancer);

  // Hot reload reducers (requires Webpack or Browserify HMR to be enabled)
  if (__DEV__ && module.hot) {
    module.hot.accept('../reducers', () =>
      // eslint-disable-next-line global-require
      store.replaceReducer(require('../reducers').default),
    );
  }

  return store;
}
