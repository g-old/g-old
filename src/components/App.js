/**
 * React Starter Kit (https://www.reactstarterkit.com/)
 *
 * Copyright Â© 2014-present Kriasoft, LLC. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { IntlProvider } from 'react-intl';
import { Provider as ReduxProvider } from 'react-redux';

const ContextType = {
  // Enables critical path CSS rendering
  // https://github.com/kriasoft/isomorphic-style-loader
  insertCss: PropTypes.func.isRequired,
  // Universal HTTP client
  fetch: PropTypes.func.isRequired,
  pathname: PropTypes.string.isRequired,
  query: PropTypes.object,
  // Integrate Redux
  // http://redux.js.org/docs/basics/UsageWithReact.html
  ...ReduxProvider.childContextTypes,
  // ReactIntl
  intl: IntlProvider.childContextTypes.intl,
  locale: PropTypes.string,
};

/**
 * The top-level React component setting context (global) variables
 * that can be accessed from all the child components.
 *
 * https://facebook.github.io/react/docs/context.html
 *
 * Usage example:
 *
 *   const context = {
 *     history: createBrowserHistory(),
 *     store: createStore(),
 *   };
 *
 *   ReactDOM.render(
 *     <App context={context}>
 *       <Layout>
 *         <LandingPage />
 *       </Layout>
 *     </App>,
 *     container,
 *   );
 */
class App extends React.PureComponent {
  static propTypes = {
    context: PropTypes.shape(ContextType).isRequired,
    children: PropTypes.element.isRequired,
  };

  static childContextTypes = ContextType;

  getChildContext() {
    return this.props.context;
  }

  componentDidMount() {
    const store = this.props.context && this.props.context.store;
    if (store) {
      this.lastLocale = store.getState().intl.locale;
      this.unsubscribe = store.subscribe(() => {
        const state = store.getState();
        const { newLocale, locale } = state.intl;
        if (!newLocale && this.lastLocale !== locale) {
          this.lastLocale = locale;
          this.forceUpdate();
        }
      });
    }
  }

  componentWillUnmount() {
    if (this.unsubscribe) {
      this.unsubscribe();
      this.unsubscribe = null;
    }
  }
  // NOTE: This methods are not needed if you update URL by setLocale action.
  //
  //  componentDidMount() {
  //    const store = this.props.context && this.props.context.store;
  //    if (store) {
  //      this.lastLocale = store.getState().intl.locale;
  //      this.unsubscribe = store.subscribe(() => {
  //        const state = store.getState();
  //        const { newLocale, locale } = state.intl;
  //        if (!newLocale && this.lastLocale !== locale) {
  //          this.lastLocale = locale;
  //          this.forceUpdate();
  //        }
  //      });
  //    }
  //  }
  //
  //  componentWillUnmount() {
  //    if (this.unsubscribe) {
  //      this.unsubscribe();
  //      this.unsubscribe = null;
  //    }
  //  }

  render() {
    const store = this.props.context && this.props.context.store;
    const state = store && store.getState();
    this.intl = (state && state.intl) || {};
    const { initialNow, locale, messages } = this.intl;
    const localeMessages = (messages && messages[locale]) || {};
    return (
      <IntlProvider
        initialNow={initialNow}
        locale={locale}
        messages={localeMessages}
        defaultLocale="de-DE"
      >
        {React.Children.only(this.props.children)}
      </IntlProvider>
    );
    /* return React.Children.only(this.props.children) */
  }
}

export default App;
