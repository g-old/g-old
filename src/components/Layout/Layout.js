/* eslint-disable react/jsx-props-no-spreading */
/**
 * React Starter Kit (https://www.reactstarterkit.com/)
 *
 * Copyright Â© 2014-present Kriasoft, LLC. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import withStyles from 'isomorphic-style-loader/withStyles';
import { defineMessages, FormattedMessage } from 'react-intl';
import { connect } from 'react-redux';
import normalizeCss from 'normalize.css';
import { createSSESub } from '../../actions/sseSubs';
import { allowCookies } from '../../actions/session';
import { setSizeVariable } from '../../actions/responsive';
import { getConsent, getLayoutSize, getSessionUser } from '../../reducers';
import { CookieLink } from '../SignUp';
import Toast from '../Toast';
import Responsive from '../../core/Responsive';
import Button from '../Button';
import Box from '../Box';

// external-global styles must be imported in your JS.
import s from './Layout.css';
import Header from '../Header';
import Footer from '../Footer';

const messages = defineMessages({
  cookieBanner: {
    id: 'cookieBanner',
    defaultMessage:
      'This site uses cookies to ensure you get the best browsing experience. More information: {link}',
    description: 'Cookie banner, should comply to EU regulations',
  },
  consent: {
    id: 'command.consent',
    defaultMessage: 'Accept',
    description: 'Commant to allow an action',
  },
});

class Layout extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
    this.consent = this.consent.bind(this);
    this.onResponsive = this.onResponsive.bind(this);
  }

  componentDidMount() {
    // eslint-disable-next-line react/destructuring-assignment
    this.props.createSSESub();
    this.responsive = Responsive.start(this.onResponsive);
  }

  componentWillUnmount() {
    if (this.responsive) {
      this.responsive.stop();
    }
  }

  onResponsive(small) {
    // eslint-disable-next-line react/destructuring-assignment
    this.props.setSizeVariable({ value: !!small });
  }

  consent() {
    // eslint-disable-next-line react/destructuring-assignment
    this.props.allowCookies();
  }

  render() {
    const { consent, children, loading, small, user } = this.props;
    let toast;
    if (!consent) {
      toast = (
        <Toast stay bottom>
          <Box align between>
            <FormattedMessage
              {...messages.cookieBanner}
              values={{ link: <CookieLink /> }}
            />
            <Button primary onClick={this.consent}>
              <FormattedMessage {...messages.consent} />
            </Button>
          </Box>
        </Toast>
      );
    }

    return (
      <div className={s.layout}>
        <div className={loading ? s.loader : null} />
        <Header small={small} user={user} />
        <div className={s.content}>{children}</div>
        <div className={s.footer}>
          {toast}
          <Footer />
        </div>
      </div>
    );
  }
}

Layout.propTypes = {
  children: PropTypes.node.isRequired,
  createSSESub: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired,
  allowCookies: PropTypes.func.isRequired,
  consent: PropTypes.string.isRequired,
  small: PropTypes.bool.isRequired,
  setSizeVariable: PropTypes.func.isRequired,
  user: PropTypes.shape({}),
};

Layout.defaultProps = {
  user: null,
};

const mapDispatch = {
  createSSESub,
  allowCookies,
  setSizeVariable,
};
const mapStateToProps = state => ({
  loading: state.ui.loading.status,
  consent: getConsent(state),
  small: getLayoutSize(state),
  user: getSessionUser(state),
});
export default connect(
  mapStateToProps,
  mapDispatch,
)(withStyles(normalizeCss, s)(Layout));
