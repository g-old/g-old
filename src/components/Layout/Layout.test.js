/**
 * React Starter Kit (https://www.reactstarterkit.com/)
 *
 * Copyright Â© 2014-present Kriasoft, LLC. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

/* eslint-env jest */
/* eslint-disable padded-blocks, no-unused-expressions */

import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { IntlProvider } from 'react-intl';
import App from '../App';
import Layout from './Layout';

const middlewares = [thunk.withExtraArgument({ fetch: () => {} })];
const mockStore = configureStore(middlewares);
const setupApp = (store, element) => (
  <IntlProvider locale="de-DE" initialNow={new Date()}>
    <App
      context={{
        intl: {
          initialNow: 1501773564065,
          formattedMessage: () => 'translated text',
          formatDate: () => 'formatted date',
          formatTime: () => 'formatted time',
          formatRelative: () => 'formatted relative',
          formatNumber: () => 'formatted number',
          formatPlural: () => 'formatted plural',
          formatMessage: () => 'formatted message',
          formatHTMLMessage: () => 'formatted html',
          now: () => new Date(),
          locale: 'de-DE',
          newLocale: null,
          messages: {
            'de-DE': {},
          },
        },
        fetch: () => {},
        insertCss: () => {},
        store,
      }}
    >
      {element}
    </App>
  </IntlProvider>
);
const initialState = {
  user: null,
  runtime: {
    initialNow: 1501773564066,
    availableLocales: ['de-DE'],
  },
  intl: {
    initialNow: 1501773564065,
    locale: 'de-DE',
    newLocale: null,
    messages: {
      'de-DE': {},
    },
  },
  entities: {
    users: {
      byId: {},
    },
    workTeams: { byId: null },
  },
  ui: {
    activityCounter: {
      feed: 0,
      proposals: 0,
    },
    loading: {
      status: false,
    },
  },
};

describe('Layout', () => {
  test('renders children correctly', () => {
    const store = mockStore(initialState);
    const wrapper = renderer
      .create(
        setupApp(
          store,
          <Layout>
            <div className="child" />
          </Layout>,
        ),
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
