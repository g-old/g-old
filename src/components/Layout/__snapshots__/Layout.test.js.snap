// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders children correctly 1`] = `
<div
  className="layout"
>
  <div
    className={null}
  />
  <span
    style={
      Object {
        "display": "none",
      }
    }
  />
  <div
    className="banner"
  >
    <img
      alt="banner"
      src="/banner01.jpg"
    />
  </div>
  <div
    className="root"
  >
    <div
      className="container"
    >
      <button
        className="button plain"
        disabled={false}
        onClick={[Function]}
      >
        <img
          alt="avatar"
          height="42px"
          src="/tile.png"
          width="42px"
        />
      </button>
      <span
        className="brand"
      >
        <span>
          G O L D
        </span>
      </span>
      <div
        role="navigation"
      >
        <div
          className="navBar"
        >
          <a
            className="link"
            href="/feed"
            onClick={[Function]}
          >
            <span>
              Feed
            </span>
          </a>
          <a
            className="link"
            href="/proposals/active"
            onClick={[Function]}
          >
            <span>
              Proposals
            </span>
          </a>
          <a
            className="link"
            href="/surveys"
            onClick={[Function]}
          >
            <span>
              Surveys
            </span>
          </a>
          <a
            className="link"
            href="/admin"
            onClick={[Function]}
          >
            <span>
              Admin
            </span>
          </a>
          <a
            className="link"
            href="/about"
            onClick={[Function]}
          >
            <span>
              About
            </span>
          </a>
        </div>
        <div
          className="menu"
        >
          <div
            className="root primary column"
            onClick={null}
          >
            <button
              className="button plain"
              disabled={false}
              onClick={[Function]}
            >
              <span
                className="icon"
              >
                <svg
                  aria-label="menu"
                  height="24px"
                  role="img"
                  version="1.1"
                  viewBox="0 0 24 24"
                  width="24px"
                >
                  <path
                    d="M2,19 L22,19 M2,5 L22,5 M2,12 L22,12"
                    fill="none"
                    stroke="#000"
                    strokeWidth="2"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="content"
  >
    <div
      className="child"
    />
  </div>
  <div
    className="footer"
  >
    <div
      className="root"
    >
      <div
        className="container"
      >
        <span
          className="text"
        >
          © G O L D
        </span>
        <span
          className="spacer"
        >
          ·
        </span>
        <a
          className="link"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
        <div>
          <span>
            <span>
              Deutsch
            </span>
             
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
